src/components/TrendContainer.js

```diff
@@ -6,51 +7,33 @@ import Loading from './Loading'
 import NoData from './NoData'
 import TrendChart from './TrendChart'
 import TrendSourceText from './TrendSourceText'
+import mungeSummaryData from '../util/summary'
 
 
-const dataByYear = data => (
-  Object.assign(
-    ...Object.keys(data).map(k => ({
-      [k]: Object.assign(...data[k].map(d => ({ [d.year]: d }))),
-    })),
-  )
-)
-
-const mungeSummaryData = (summaries, ucr, place) => {
-  if (!summaries || !summaries[place]) return false
-
-  const keys = Object.keys(summaries)
-  const summaryByYear = dataByYear(summaries)
-  const ucrByYear = dataByYear(ucr)
-
-  return summaries[place].map(d => (
-    Object.assign(
-      { date: d.year },
-      ...keys.map(k => {
-        const count = summaryByYear[k][d.year].actual
-        const pop = ucrByYear[k][d.year].total_population
-        return { [k]: { count, pop, rate: (count / pop) * 100000 } }
-      }),
-    )
-  ))
+const filterDataWithinYears = ({ data, since, until }) => {
```

Comment:
i think this could do the trick:

```
const mungeSummaryData = (crime, summaries, place, since, until) => {
  if (!summaries || !summaries[place]) return false

  const keys = Object.keys(summaries)
  return summaries[place]
    .filter(d => d.year >= since && d.year <= until)
    .map(year => {
      const data = { date: year.year }
      keys.forEach(key => {
        const source = key !== place
          ? summaries[key].find(d => d.year === data.date)
          : year
        data[key] = {
          pop: source.population,
          count: source[crime],
          rate: source[crime] / source.population * 100000,
        }
      })
      return data
    })
}
```

note the `filter` call before the map (and new `since` and `until` params)